packages/typescript-estree/tests/lib/convert.test.ts

```diff
@@ -449,5 +449,79 @@ describe('convert', () => {
       expect(tsMappedType.typeParameter).toBeUndefined();
       expect(Object.keys(tsMappedType)).toContain('typeParameter');
     });
+
+    describe('TSMappedType AST shape consistency', () => {
```

Comment:
[Testing] We normally put fixtures in `packages/ast-spec/src`. Is there a reason not to here?

If not, I think `packages/ast-spec/src/type/TSMappedType` would be the right place to add new fixtures around `readonly` and/or optional mapped types.